source/blender/makesrna/intern/rna_sensor.c

```diff
@@ -867,41 +867,67 @@ static void rna_def_joystick_sensor(BlenderRNA *brna)
 	PropertyRNA *prop;
 
 	static EnumPropertyItem event_type_joystick_items[] = {
-		{SENS_JOY_BUTTON, "BUTTON", 0, "Button", ""},
-		{SENS_JOY_AXIS, "AXIS", 0, "Axis", ""},
-		{SENS_JOY_HAT, "HAT", 0, "Hat", ""},
-		{SENS_JOY_AXIS_SINGLE, "AXIS_SINGLE", 0, "Single Axis", ""},
+	    {SENS_JOY_AXIS, "STICK_DIRECTIONS", 0, "Stick Directions", ""},
+	    {SENS_JOY_AXIS_SINGLE, "STICK_AXIS", 0, "Stick Axis", ""},
+	    {SENS_JOY_SHOULDER_TRIGGER, "SHOULDER_TRIGGERS", 0, "Shoulder Triggers", ""},
+		{SENS_JOY_BUTTON, "BUTTONS", 0, "Buttons", ""},
 		{0, NULL, 0, NULL, NULL}
 	};
 
+	static EnumPropertyItem axis_items[] = {
+	    {SENS_JOY_LEFT_STICK, "LEFT_STICK", 0, "Left Stick", ""},
+	    {SENS_JOY_RIGHT_STICK, "RIGHT_STICK", 0, "Right Stick", ""},
+	    {0, NULL, 0, NULL, NULL}
+	};
+
 	static EnumPropertyItem axis_direction_items[] = {
-		{SENS_JOY_X_AXIS, "RIGHTAXIS", 0, "Right Axis", ""},
-		{SENS_JOY_Y_AXIS, "UPAXIS", 0, "Up Axis", ""},
-		{SENS_JOY_NEG_X_AXIS, "LEFTAXIS", 0, "Left Axis", ""},
-		{SENS_JOY_NEG_Y_AXIS, "DOWNAXIS", 0, "Down Axis", ""},
+		{SENS_JOY_X_AXIS, "RIGHTAXIS", 0, "Right", ""},
+		{SENS_JOY_Y_AXIS, "UPAXIS", 0, "Up", ""},
+		{SENS_JOY_NEG_X_AXIS, "LEFTAXIS", 0, "Left", ""},
+		{SENS_JOY_NEG_Y_AXIS, "DOWNAXIS", 0, "Down", ""},
 		{0, NULL, 0, NULL, NULL}
 	};
 
-	static EnumPropertyItem hat_direction_items[] = {
-		{SENS_JOY_HAT_UP, "UP", 0, "Up", ""},
-		{SENS_JOY_HAT_DOWN, "DOWN", 0, "Down", ""},
-		{SENS_JOY_HAT_LEFT, "LEFT", 0, "Left", ""},
-		{SENS_JOY_HAT_RIGHT, "RIGHT", 0, "Right", ""},
-
-		{SENS_JOY_HAT_UP_RIGHT, "UPRIGHT", 0, "Up/Right", ""},
-		{SENS_JOY_HAT_DOWN_LEFT, "DOWNLEFT", 0, "Down/Left", ""},
-		{SENS_JOY_HAT_UP_LEFT, "UPLEFT", 0, "Up/Left", ""},
-		{SENS_JOY_HAT_DOWN_RIGHT, "DOWNRIGHT", 0, "Down/Right", ""},
+	static EnumPropertyItem axis_trigger_items[] = {
+	    {SENS_JOY_LEFT_SHOULDER_TRIGGER, "LEFT_SHOULDER_TRIGGER", 0, "Left Shoulder Trigger", ""},
```

Comment:
I guess you have an issue with the indentation, it was two tabs before and it's a tabs + 4 spaces here. In the code we see nothing as a tab is 4 spaces, but it can show that your editor is using only spaces and that all the code is with spaces indentation. @lordloki: Can you check that, and replace with tabs if it's possible ?